src/plugins/legacy/polygonRoi/polygonEventFilter.cpp

```diff
@@ -68,145 +69,137 @@ void polygonEventFilter::updateView(medAbstractImageView *view)
     currentView = view;
 }
 
-bool polygonEventFilter::mouseReleaseEvent(medAbstractView *view, QMouseEvent *mouseEvent)
+void polygonEventFilter::reset()
 {
-    bool bRes = false;
-
-    if (isRepulsorActivated)
+    for (medTagRoiManager *manager: managers)
     {
-        if (interactorStyleRepulsor != nullptr)
-        {
-            interactorStyleRepulsor->GetManager()->SetMasterRoi(true);
-            interactorStyleRepulsor->GetManager()->interpolateIfNeeded();
-            bRes = true;
-        }
+        delete manager;
+    }
+    managers.clear();
+    cursorState = CURSORSTATE::CS_MOUSE_EVENT;
+    activateRepulsor(false);
+    if (interactorStyleRepulsor)
+    {
+        interactorStyleRepulsor->Delete();
+        interactorStyleRepulsor = nullptr;
     }
-
-    return bRes;
 }
 
-void polygonEventFilter::removeManagers()
+bool polygonEventFilter::eventFilter(QObject *obj, QEvent *event)
 {
-    for (medLabelManager *manager: managers)
+    QKeyEvent *keyEvent = dynamic_cast<QKeyEvent*>(event);
+    QMouseEvent *mouseEvent = dynamic_cast<QMouseEvent*>(event);
+
+    if (mouseEvent)
```

Comment:
some {} in your changes in this file to add if possible

Reply:
done